src/main/java/fi/metatavu/kuntaapi/server/controllers/IdentifierController.java

```diff
@@ -0,0 +1,627 @@
+import fi.metatavu.kuntaapi.server.id.ServiceLocationServiceChannelId;
+import fi.metatavu.kuntaapi.server.id.ShortlinkId;
+import fi.metatavu.kuntaapi.server.id.TileId;
+import fi.metatavu.kuntaapi.server.id.WebPageServiceChannelId;
+import fi.metatavu.kuntaapi.server.integrations.KuntaApiConsts;
+import fi.metatavu.kuntaapi.server.integrations.KuntaApiIdFactory;
+import fi.metatavu.kuntaapi.server.persistence.dao.ArchivedIdentifierDAO;
+import fi.metatavu.kuntaapi.server.persistence.dao.IdentifierDAO;
+import fi.metatavu.kuntaapi.server.persistence.dao.IdentifierRelationDAO;
+import fi.metatavu.kuntaapi.server.persistence.model.ArchivedIdentifier;
+import fi.metatavu.kuntaapi.server.persistence.model.Identifier;
+import fi.metatavu.kuntaapi.server.persistence.model.IdentifierOrderIndex;
+import fi.metatavu.kuntaapi.server.persistence.model.IdentifierRelation;
+
+/**
+ * Identifier controller
+ * 
+ * @author Antti Leppä
+ */
+@ApplicationScoped
+@SuppressWarnings ("squid:S3306")
+public class IdentifierController {
+  
+  @Inject
+  private Logger logger;
+  
+  @Inject
+  private IdentifierDAO identifierDAO;
+  
+  @Inject
+  private ArchivedIdentifierDAO archivedIdentifierDAO;
+  
+  @Inject
+  private IdentifierRelationDAO identifierRelationDAO;
+  
+  @Inject
+  private KuntaApiIdFactory kuntaApiIdFactory;
+  
+  public Identifier findIdentifierById(BaseId id) {
+    if (id == null) {
+      logger.severe("Passed null id to findIdentifierById method");
+      return null;
+    }
+    
+    String organizationKuntaApiId = null;
+    if (id instanceof OrganizationBaseId) {
+      organizationKuntaApiId = getOrganizationBaseIdOrganizationKuntaApiId((OrganizationBaseId) id);
+    }
+    
+    return findIdentifierByTypeSourceAndIdOrganizationId(id.getType(), id.getSource(), id.getId(), organizationKuntaApiId);
+  }
+  
+  /**
+   * Acquires an identifier. 
+   * 
+   * 1) If identifier exists and is not archived, the identified is updated and returned
+   * 2) If identifier is archived, the identifier is unarchived and returned
+   * 3) If identifier does not exist, new identifier is created and returned
+   * 
+   * @param orderIndex orderIndex of index
+   * @param id id
+   * @return identifier
+   */
+  public Identifier acquireIdentifier(Long orderIndex, BaseId id) {
+    Identifier identifier = findIdentifierById(id);
+    if (identifier == null) {
+      ArchivedIdentifier archivedIdentifier = findArchivedIdentifierById(id);
+      if (archivedIdentifier != null) {
+        identifier = createIdentifier(orderIndex == null ? archivedIdentifier.getOrderIndex() : orderIndex, archivedIdentifier.getType(), archivedIdentifier.getKuntaApiId(), archivedIdentifier.getSource(), archivedIdentifier.getSourceId(), archivedIdentifier.getOrganizationKuntaApiId());
+        archivedIdentifierDAO.delete(archivedIdentifier);
+      } else {
+        identifier = createIdentifier(orderIndex == null ? 0l : orderIndex, id);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') "NullPointerException" will be thrown when invoking method "createIdentifier()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)